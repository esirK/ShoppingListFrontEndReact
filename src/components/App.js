import React, { Component } from 'react';
import RegisterForm from '../containers/register_form'

export default class App extends Component {
  render() {
    return (
      <div className="container">
        <RegisterForm />
      </div>
    );
  }
}