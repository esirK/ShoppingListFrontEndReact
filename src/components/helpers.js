export function validate(values){
	const errors = {};
	if (!values.username){
		errors.username = 'Username Required';
	}
	if (!values.email){
		errors.email = 'Email Required';
	}
	if (!values.password){
		errors.password = 'Password Required';
	}
	if (values.password && values.password.length < 6){
		errors.password = 'Passwords must be at least 6 Characters Long';
	}
	if (!values.confirm_password){
		errors.confirm_password = 'Enter confirmation Password';
	}
	if(values.password !== values.confirm_password){
		console.log('Look at me ',values.confirm_password);
		errors.confirm_password = 'Passwords do not match';
	}
	return errors;
}
