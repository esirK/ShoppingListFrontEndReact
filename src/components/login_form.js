import React, {Component} from 'react';


export default class LoginForm extends Component{
    render(){
        return(
            <div>
                <p>I am Mr Meesix Look at me</p>
            </div>
        );
    }
}