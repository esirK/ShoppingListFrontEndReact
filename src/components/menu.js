import React, {Component} from 'react';
import {connect} from 'react-redux';

import MenuItem from 'material-ui/MenuItem';
import TextField from 'material-ui/TextField';
import {FlatButton} from 'material-ui';

import Dialog from 'material-ui/Dialog';
import Snackbar from 'material-ui/Snackbar';

import IconMenu from 'material-ui/IconMenu';
import IconButton from 'material-ui/IconButton';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';

import {Logout, update_profile, hideSnackBar} from '../actions';

export class Menu extends Component{
	constructor(props){
		super(props);
		this.state ={
			conf_logout: false,
			open_profile: false,
			username: '',
			password: '',
			old_password: '',
			openSb: false
		};
		this.logout = this.logout.bind(this);
		this.profile = this.profile.bind(this);
		this.handleClose = this.handleClose.bind(this);
		this.handleLogout = this.handleLogout.bind(this);
		this.handlePasswordChange = this.handlePasswordChange.bind(this);
		this.handleUsernameChange = this.handleUsernameChange.bind(this);
		this.handleProfileUpdate = this.handleProfileUpdate.bind(this);
	}
	logout(){
		this.setState({conf_logout: true});
	}
	profile(){
		this.setState({open_profile: true});
		this.props.hideSnackBar();
	}
	handleLogout(){
		this.props.Logout();
		this.setState({conf_logout: false});
		//reload page on Logout
		window.location.reload();
	}
	handlePasswordChange(e){
		this.setState({password: e.target.value});
	}
	handleUsernameChange(e){
		this.setState({username: e.target.value});
	}
	handleProfileUpdate(){
		this.props.update_profile({username:this.state.username, password:this.state.password});
		this.setState({open_profile: false});
		this.setState({'username':''});
		this.setState({'password':''});
		this.setState({'old_password':''});
	}
	handleClose(){
		this.setState({conf_logout: false});
		this.setState({open_profile: false});
		this.setState({'username':''});
		this.setState({'password':''});
		this.setState({'old_password':''});
		this.props.hideSnackBar();
	}
	render(){
		let actions = [
			<FlatButton
				label="Cancel"
				primary={true}
				onClick={this.handleClose}
			/>,
			<FlatButton
				label="OK"
				primary={true}
				onClick={this.handleLogout}
			/>,
		];
		let profile_actions = [
			<FlatButton
				label="Cancel"
				primary={true}
				onClick={this.handleClose}
			/>,
			<FlatButton
				label="Update"
				primary={true}
				onClick={this.handleProfileUpdate}
			/>,
		];
		const profile_children = [
			<TextField
			  hintText="New Username"
			  value={this.state.username}
			  fullWidth={true}
			  key='name'
			  onChange={this.handleUsernameChange}
			/>,
			<TextField
			  hintText="Old Password"
			  multiLine={true}
			  fullWidth={true}
			  value={this.state.password}
			  key='old_p'
			  onChange={this.handlePasswordChange}
			  type={'password'}
			/>,
			<TextField
			  hintText="New Password"
			  multiLine={true}
			  fullWidth={true}
			  key='new_p'
			  onChange={this.handlePasswordChange}
			  type="password"
			>
			</TextField>
		];
		return(
			<div>
				<IconMenu
					iconButtonElement={
						<IconButton><MoreVertIcon /></IconButton>
					}
					targetOrigin={{horizontal: 'left', vertical: 'top'}}
					anchorOrigin={{horizontal: 'right', vertical: 'top'}}
				>
					<MenuItem primaryText="Profile" onClick={this.profile} />
					<MenuItem primaryText="Sign out" onClick={this.logout}/>
				</IconMenu>
				<Dialog
					actions={actions}
					modal={false}
					open={this.state.conf_logout}
					onRequestClose={this.handleClose}
				>
         			Confirm Logout?
				</Dialog>
				<Dialog open={this.state.open_profile}
					title="Update Personal details"
					actions={profile_actions}
					children={profile_children}
					onRequestClose={this.handleClose}
				>
				</Dialog>
				<Snackbar
					open={this.props.openSb}
					message={this.props.message}
					autoHideDuration={3000}
					onRequestClose={this.handleRequestClose}
				/>
			</div>
		);
	}
}
function mapStateToProps(state) {
	return{
		message: state.authentication.message,
		openSb: state.authentication.openSb,
	};
}
export default connect(mapStateToProps, {Logout, update_profile, hideSnackBar})(Menu);