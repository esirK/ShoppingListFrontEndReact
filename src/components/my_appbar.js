import React, { Component } from 'react';

import {connect} from 'react-redux';
import AppBar from 'material-ui/AppBar';

import Menu from './menu';

export class MyAppBar extends Component{
	render(){
		return(
			<AppBar title="Shopping List App" showMenuIconButton={false}
				iconElementRight={this.props.isAuthenticated?<Menu />:null}
			/>
		);
	};
}
function mapStateToProps(state, ownProps) {
	return{
		isAuthenticated: state.authentication.isAuthenticated,
	};
}
export default connect(mapStateToProps, null)( MyAppBar);