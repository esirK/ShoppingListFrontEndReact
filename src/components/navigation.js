import React, { Component } from 'react';
import { Link } from 'react-router-dom';


export default class Navigation extends Component{
	render(){
		return(
			<div className="container">
				<div className="navbar-header">
					<a className="navbar-brand" href="/" title="Awesome Shopping list app">

					</a>
				</div>
				<div className="navbar-collapse ">
					<ul>
						<li>
							<Link to="/">
                                Home
							</Link>
						</li>
					</ul>
				</div>
			</div>
		);
	}
}