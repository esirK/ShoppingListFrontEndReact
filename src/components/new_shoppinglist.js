import React, {Component} from 'react';
import {connect} from 'react-redux';

import Dialog from 'material-ui/Dialog';
import CircularProgress from 'material-ui/CircularProgress';


import {itemActions, newShoppinglistChildren} from './helpers';
import {addNewShoppingList, resetErrors, closeFab} from '../actions';

import ShoppingLists from './shoppinglists';
export class AddShoppingList extends Component{

	constructor(props){
		super(props);
		this.state = {
			name: '',
			description: '',
			exit: false,
			open: true,
		};
		this.handleClose = this.handleClose.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
		this.handleTitleChange = this.handleTitleChange.bind(this);
		this.handleDescChange = this.handleDescChange.bind(this);
	}
	handleClose(){
		// resets state on closeing of Add shoppinglist dialog
		this.setState({ name: '' });
		this.setState({ description: ''});
		this.setState({ exit: true });
		this.props.closeFab();
		this.props.resetErrors();
	}
	handleSubmit(){
		/*Submits user request to create 
		a new shoppinglist with the required parameters*/
		let limit = this.props.limit;
		this.props.addNewShoppingList({
			'name': this.state.name,
			'description': this.state.description
		}, limit);
		this.setState({ name: '' });
		this.setState({ description: ''});
	}
	handleTitleChange(e){
		// Updates title field as user types
		this.setState({ name: e.target.value });
	}
	handleDescChange(e){
		// Updates description as user types
		this.setState({ description: e.target.value });
	}
	render()
	{
		const actions = itemActions(this);

		if(this.props.isCreatingNewShoppingList)
		{
			//add a circular progress bar bettween the 2 buttons
			actions.splice(1, 0, <CircularProgress/>);
		}
		const children = newShoppinglistChildren(this);
		//if the shoppinglist has been created then return to parent
		if(this.props.shoppinglistCreated | this.state.exit){
			return(
				<ShoppingLists/>
			);  
		}
		return(
			<Dialog open={this.state.open}
				title="Add Shoppinglist"
				actions={actions}
				children={children}
			>
			</Dialog>
		);
	};
}
function mapStateToProps(state){
	return {
		isCreatingNewShoppingList: state.shoppinglists.isCreatingNewShoppingList,
		error: state.shoppinglists.error,
		message: state.shoppinglists.message,
		shoppinglistCreated: state.shoppinglists.shoppinglistCreated,
	};
}
export default connect(mapStateToProps, {addNewShoppingList, closeFab,
	resetErrors}) (AddShoppingList);