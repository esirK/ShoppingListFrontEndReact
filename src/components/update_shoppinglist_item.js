import React, {Component} from 'react';

import {connect} from 'react-redux';

import {deactivateUpdateItem, viewShoppingList, updateShoppingListItem, resetErrors} from '../actions';
import {itemActions, updateChildren} from './helpers';

import Dialog from 'material-ui/Dialog';

class UpdateShoppingListItem extends Component{
	constructor(props){
		super(props);
		this.state = {
			name: props.item.name,
			price: props.item.price,
			quantity: props.item.quantity,
			nameError: '',
			priceError: '',
			quantityError: '',
			error:''
		};
		this.handleNameChange = this.handleNameChange.bind(this);
		this.handlePriceChange = this.handlePriceChange.bind(this);
		this.handleQuantityChange = this.handleQuantityChange.bind(this);
		this.handleClose = this.handleClose.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}
	handleNameChange(e){
		this.setState({ name: e.target.value });
		//Update the name field
		if ((e.target.value).trim().length===0){
			this.setState({
				nameError: 'Name is Empty'
			});
		}
		else if(!isNaN(this.state.name)){
			this.setState({nameError: 'Only Strings allowed For shoppinglist names'});
		}
		else{
			this.setState({
				nameError: ''
			});
		}
	}
	handlePriceChange(e){
		//Update the Price field
		this.setState({ price: e.target.value });
		if ((e.target.value).trim().length===0){
			this.setState({
				priceError: 'Price is Empty'
			});
		}
		else if(isNaN(this.state.price)){
			this.setState({priceError: 'Only Numbers allowed For shoppinglist price'});
		}
		else{
			this.setState({
				priceError: ''
			});
		}
	}
	handleQuantityChange(e){
		//Update the Quantity field
		this.setState({ quantity: e.target.value });
		if ((e.target.value).trim().length===0){
			this.setState({
				quantityError: 'Quantity is Empty'
			});
		}
		else if(isNaN(this.state.quantity)){
			this.setState({quantityError: 'Only Numbers allowed For shoppinglist quantity'});
		}
		else{
			this.setState({
				quantityError: ''
			});
		}
	}
	handleClose(){
		//Invoked when the cancle button is clicked
		//Sets the openAddItem props to false
		this.props.deactivateUpdateItem();
		this.props.resetErrors();
	}
	handleSubmit(){
		//Invoked When a User presses the create new shoppinglist item button
		//submit details Supplied
		if(this.state.nameError !==''&&this.state.priceError!==''&& this.state.quantityError!==''){
			this.setState({error:'Nothing Provided to Update'});
		}
		if(this.state.nameError ===''&&this.state.priceError===''&& this.state.quantityError==='')
		{
			this.props.updateShoppingListItem({
				'name': this.state.name,
				'price': this.state.price,
				'quantity': this.state.quantity,
				'item_id': this.props.item.id
			});
		}
	}
	render(){
		const actions = itemActions(this);
		const children = updateChildren(this);
		return(
			<div>
				<Dialog open={this.props.openUpdateItem}
					title="Update Shoppinglist Item"
					actions={actions}
					children={children}
				>
				</Dialog>
			</div>
		);
	}
}
function mapStateToProps(state){
	return{
		openUpdateItem: state.shoppinglist_items.isOpenUpdateItem,
		isUpdating: state.shoppinglist_items.isUpdating,
		error: state.shoppinglist_items.error,
	};
}
export default connect(mapStateToProps, {deactivateUpdateItem, viewShoppingList, updateShoppingListItem, resetErrors})(UpdateShoppingListItem);