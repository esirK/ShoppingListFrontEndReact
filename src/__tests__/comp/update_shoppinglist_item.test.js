import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import {UpdateShoppingListItem} from '../../components/update_shoppinglist_item';

describe('Render Update Shopping List items',()=>{
	const props = {
		item:{
			id:'32',
			name:'Monster',
			price:'200.0',
			quantity:'2.5'
		},
		openUpdateItem:true
	};
    
	const shoppinglistItems = shallow(<UpdateShoppingListItem {...props}/>);
	it('Renders UpdateShoppingListItem component', ()=>{
		expect(shallowToJson(shoppinglistItems)).toMatchSnapshot();
	});
});