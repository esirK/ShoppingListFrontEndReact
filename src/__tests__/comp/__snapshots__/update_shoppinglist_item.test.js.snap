// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Update Shopping List items Renders UpdateShoppingListItem component 1`] = `
<div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
        />,
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Submit"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
        />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={false}
    open={true}
    repositionOnUpdate={true}
    title="Update Shoppinglist Item"
  >
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      fullWidth={true}
      hintText="New Item Name"
      key="name"
      multiLine={false}
      onChange={[Function]}
      rows={1}
      type="text"
      underlineShow={true}
      value="Monster"
    />
    <TextField
      disabled={false}
      errorText=""
      floatingLabelFixed={false}
      fullWidth={true}
      hintText="New Price of the Item"
      key="price"
      multiLine={true}
      onChange={[Function]}
      rows={1}
      type="text"
      underlineShow={true}
      value="200.0"
    />
    <TextField
      disabled={false}
      errorText=""
      floatingLabelFixed={false}
      fullWidth={true}
      hintText="Quantity of the Item"
      key="quantity"
      multiLine={true}
      onChange={[Function]}
      rows={1}
      type="text"
      underlineShow={true}
      value="2.5"
    />
    <p
      className="alert alert-danger"
      key="error"
    />
  </Dialog>
</div>
`;
